### ISSUE TITLE
Does cropping the output of a type-1 nufft create an array copy?

### ISSUE THREAD

**Issue by michael-0brien** [flatironinstitute/jax-finufft#202]
Hello! In my use-case of `jax-finufft`, I compute a type-1 nufft of real-valued weights and then crop it to the half plane for better performance. I'd like to understand if this operation creates an extra array copy. Here is a minimal reproducible example:

```python
import functools as ft, jax, numpy as np, jax.numpy as jnp
import jax_finufft
from jax_finufft.options import Opts

@ft.partial(jax.jit, static_argnums=(2,))
def f(xy, c, dim):
    # Compute fftn grid at shape `(10, 10)`
    x, y = xy
    out_fftn = jax_finufft.nufft1((dim, dim), c, x, y, opts=Opts(modeord=True))
    # Crop to rfftn shape (a naive implementation of this; in reality also have to
    # enforce that self-conjugate components are real-valued)
    out_rfftn = out_fftn[:, : dim // 2 + 1]
    return out_rfftn

dim, npts = 10, 100
x, y = (np.random.uniform(-np.pi, np.pi, size=npts), np.random.uniform(-np.pi, np.pi, size=npts))
c = np.ones_like(x, dtype=complex)
print(f.lower((x, y), c, 10).compile().as_text())
```

And the corresponding HLO dump

```
HloModule jit_f, is_scheduled=true, entry_computation_layout={(f32[100]{0}, f32[100]{0}, c64[100]{0})->c64[10,6]{1,0}}, allow_spmd_sharding_propagation_to_parameters={true,true,true}, allow_spmd_sharding_propagation_to_output={true}

%fused_computation (param_0.1: c64[1,1,10,10]) -> c64[10,6] {
  %param_0.1 = c64[1,1,10,10]{3,2,1,0} parameter(0)
  %bitcast.4 = c64[10,10]{1,0} bitcast(%param_0.1), metadata=...
  ROOT %slice.0 = c64[10,6]{1,0} slice(%bitcast.4), slice={[0:10], [0:6]}, metadata=...
}

ENTRY %main.2 (xy_0_.1: f32[100], xy_1_.1: f32[100], c.1: c64[100]) -> c64[10,6] {
  %xy_0_.1 = f32[100]{0} parameter(0), metadata={op_name="xy[0]"}
  %xy_1_.1 = f32[100]{0} parameter(1), metadata={op_name="xy[1]"}
  %c.1 = c64[100]{0} parameter(2), metadata={op_name="c"}
  %constant.0 = u8[144]{0} constant({...}), metadata={op_name="jit(f)/jit(nufft1)"}
  %bitcast.2 = f32[1,100]{1,0} bitcast(%xy_0_.1), metadata={op_name="xy[0]"}
  %bitcast.1 = f32[1,100]{1,0} bitcast(%xy_1_.1), metadata={op_name="xy[1]"}
  %bitcast = c64[1,1,100]{2,1,0} bitcast(%c.1), metadata={op_name="c"}
  %nufft1.0 = c64[1,1,10,10]{3,2,1,0} custom-call(%constant.0, %bitcast, %bitcast.1, %bitcast.2), custom_call_target="nufft2d1f", operand_layout_constraints={u8[144]{0}, c64[1,1,100]{2,1,0}, f32[1,100]{1,0}, f32[1,100]{1,0}}, api_version=API_VERSION_STATUS_RETURNING, metadata=...
}
```

A few questions / comments (some of which may be appropriate for a JAX discussions thread, but wanted to start here):

1) I am no expert on reading HLO dumps! Does anyone have experience reading these to understand which conditions copies are created?
2) I believe my question is general for any computation in JAX where the compiler cannot fuse the computation and eliminate the creation of the `(10, 10)` array all together.
3) Does anyone have insight on if it makes a difference if the line `out_fftn[:, : dim // 2 + 1]` is replaced by `out_fftn.at[:, : dim // 2 + 1].get()`?
